core/chains/evm/gas/rollups/op_l1_oracle.go

```diff
@@ -34,53 +34,64 @@ type OptimismL1Oracle struct {
 	logger     logger.SugaredLogger
 	chainType  chaintype.ChainType
 
-	l1OracleAddress     string
-	gasPriceMethod      string
-	l1GasPriceMethodAbi abi.ABI
-	l1GasPriceMu        sync.RWMutex
-	l1GasPrice          priceEntry
-
-	gasCostMethod      string
-	l1GasCostMethodAbi abi.ABI
+	l1OracleAddress string
+	l1GasPriceMu    sync.RWMutex
+	l1GasPrice      priceEntry
+	isEcotone       bool
+	isFjord         bool
+	upgradeCheckTs  int64
 
 	chInitialised chan struct{}
 	chStop        services.StopChan
 	chDone        chan struct{}
 
-	isEcotoneMethodAbi abi.ABI
-
-	l1BaseFeeCalldata    []byte
-	isEcotoneCalldata    []byte
-	getL1GasUsedCalldata []byte
-	getL1FeeCalldata     []byte
-
-	isEcotone        bool
-	isEcotoneCheckTs int64
+	getL1FeeMethodAbi         abi.ABI
+	l1BaseFeeCalldata         []byte
+	baseFeeScalarCalldata     []byte
+	blobBaseFeeCalldata       []byte
+	blobBaseFeeScalarCalldata []byte
+	decimalsCalldata          []byte
+	isEcotoneCalldata         []byte
+	isFjordCalldata           []byte
 }
 
 const (
-	// OPStackGasOracle_isEcotone fetches if the OP Stack GasPriceOracle contract has upgraded to Ecotone
-	OPStackGasOracle_isEcotone = "isEcotone"
-	// OPStackGasOracle_getL1GasUsed fetches the l1 gas used for given tx bytes
-	OPStackGasOracle_getL1GasUsed = "getL1GasUsed"
-	// OPStackGasOracle_isEcotonePollingPeriod is the interval to poll if chain has upgraded to Ecotone
+	// isUpgradedPollingPeriod is the interval to poll if chain has been upgraded
 	// Set to poll every 4 hours
-	OPStackGasOracle_isEcotonePollingPeriod = 14400
-	// OPStackGasOracleAddress is the address of the precompiled contract that exists on OP stack chain.
-	// OPStackGasOracle_l1BaseFee fetches the l1 base fee set in the OP Stack GasPriceOracle contract
-	// OPStackGasOracle_l1BaseFee is a hex encoded call to:
-	// `function l1BaseFee() external view returns (uint256);`
-	OPStackGasOracle_l1BaseFee = "l1BaseFee"
-	// OPStackGasOracle_getL1Fee fetches the l1 fee for given tx bytes
-	// OPStackGasOracle_getL1Fee is a hex encoded call to:
+	upgradePollingPeriod = 14400
```

Comment:
nit: perhaps we could change this to a more meaningful `4 * time.Hour` type and use builtin `time.Since` method when checking for last update. Same goes for `upgradeCheckTs`. It's better if we change it to: `upgradeCheckTs  time.Time`